airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -481,13 +485,21 @@ def request_params(
 
         stream_date = stream_state.get(self.cursor_field)
         next_token = (next_page_token or {}).get("next_token")
+        primary_key = (next_page_token or {}).get("primary_key")
         start_date = next_token or stream_date or self.start_date
+        self.prev_start_date = start_date
 
         query = f"SELECT {','.join(selected_properties.keys())} FROM {self.name} "
         if start_date:
-            query += f"WHERE {self.cursor_field} >= {start_date} "
+            if primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
```

Comment:
I understand why we need to check `primary_key` here, but what is the purpose of `self.name not in UNSUPPORTED_FILTERING_STREAMS` as part of the condition? Is this intended to dictate whether or not the stream supports using a WHERE clause in query?

This block feels very confusing to read with a lot of repeated conditions which makes it a little hard to reason about and I'm trying to think of ways we can streamline this query builder a bit

Reply:
Yes, good point. I agree it's not very obvious.

In really I just try to be consistent with this part of code, look at master branch:

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L489

Because my "improved" query strictly relaies on `ORDER BY`
if `ORDER BY` is not used I also skip my "improved" query

Reply:
@brianjlai Searching the documentation, some of those UNSUPPORTED_FILTERING_STREAMS fields support only the "where" filter which is consistent with the master logic. Though, I am not sure about it (since for those fields without the "order" the next date can be random ?).

I tried to find any refactors to the code to maintain that consistency but failed to do so.

Reply:
> I tried to find any refactors to the code to maintain that consistency but failed to do so.

We could add a `supports_filtering` attribute to `BulkIncrementalSalesforceStream`  and set this attribute on instantiation after checking `if name in UNSUPPORTED_FILTERING_STREAMS`